sunpy/instr/rhessi.py

```diff
@@ -348,3 +400,185 @@ def _build_energy_bands(label, bands):
+    result = result[result['START_TIME'] < end_dt]
+    return result
+
+
+def read_flare_file(file):
+    """
+    Read RHESSI flare list .fits file into ``pandas.DataFrame``. TIME values are parsed with
+    format 'utime', which is the same as Unix timestamp but starts 9 years later.
+    FLAGS are assigned their respective label (FLAG ID) and returned as `dict`.
+
+    Parameters
+    ----------
+    file : `str`
+        The URL or local filename of the hessi flare list.
+
+    Returns
+    -------
+    ``pandas.DataFrame``
+        out : ``pandas.DataFrame`` containing the parsed flares.
+
+    Examples
+    --------
+    >>> from sunpy.instr.rhessi as read_flare_file as rff
+    >>> rff("https://hesperia.gsfc.nasa.gov/hessidata/dbase/hessi_flare_list_201802.fits")
+
+    References
+    ----------
+    https://hesperia.gsfc.nasa.gov/rhessi3/data-access/rhessi-data/flare-list/index.html
+    """
+    try:
+        fits = sunpy.io.fits.read(file)
+    except Exception:
+        raise RuntimeError("couldn't load file " + file)
+
+    results = []
+    for row in fits[3].data[0:20]:
+        result_row = {}
+        for k in fits_rec_keys:
+            if k.endswith('_TIME'):
+                result_row[k] = parse_time(row[k], format="utime")
+                result_row[k].format = "datetime"  # for human readable display inside the DF
+            elif k == 'FLAGS':
+                flags = {}
+                for i, fid in zip(row[k], fits[2].data['FLAG_IDS'][0]):
+                    flags[fid] = i
+                result_row[k] = flags
+            else:
+                result_row[k] = row[k]
+        results.append(result_row)
+
+    return pd.DataFrame(results)
+
+
+def print_flare_list(data_frame):
+    """
+    Convert and print flares similar to the available .txt flare lists.
+
+    Parameters
+    ----------
+    data_frame : ``pandas.DataFrame``
+        DataFrame containing the flares to print.
+    """
+
+    for idx, row in data_frame.iterrows():
+        print(
+            "{id:9} {st} {pt} {et} {dur:5} {p:6} {n:9} {e:>11} {x:5} {y:5} {r:6} {ar:4}  {f}"
+            .format(
+                id=row['ID_NUMBER'],
+                st=row['START_TIME'].strftime('%e-%b-%Y %H:%M:%S'),
+                pt=row['PEAK_TIME'].strftime('%H:%M:%S'),
+                et=row['END_TIME'].strftime('%H:%M:%S'),
+                dur=int(round((row['END_TIME'] - row['START_TIME']).to_value("sec"))),
+                p=int(row['PEAK_COUNTRATE']),
+                n=int(row['TOTAL_COUNTS']),
+                e=str(int(row['ENERGY_HI'][0])) + "-" + str(int(row['ENERGY_HI'][1])),
+                x=int(round(row['POSITION'][0])),
+                y=int(round(row['POSITION'][1])),
+                r=int(round((row['POSITION'][0] ** 2 + row['POSITION'][1] ** 2) ** 0.5)),
+                ar=row['ACTIVE_REGION'],
+                f=" ".join(convert_flag_dict(row['FLAGS'])),
+            )
+        )
+
+
+def convert_flag_dict(flags_dict):
```

Comment:
Not sure.. I still "made it public" as it could be helpful if you want to do something else with it than print it directly. The raw flags are quite messy.. but yeah, its a rather rare use case i guess

Reply:
I am not sure if I am keen of these two functions if they have a rare use case. Willing to be persuaded otherwise.

Reply:
Do I understand correct, you would return both the raw and the formatted flags in the DF?
I think there are 1 or 2 flags that are not present in the formatted version, which is why I think access to the raw flags should not be lost. However they also don't seem to be of much interest to the people that use this data, otherwise it would be included in the .txt version of the flare lists...

Reply:
Sorry, should've clarified. Yes, I meant return both the raw and formatted flags so that the raw data is not lost and so that the user does not have to call an additional function to get the formatted ones. I have not worked with this data at all so others with more experience may have different opinions.